src/feature/hs/hs_service.c

```diff
@@ -3404,6 +3404,29 @@ service_encode_descriptor(const hs_service_t *service,
   return ret;
 }
 
+/* An introduction circuit has timed out while building or waiting for an
+ * established introduction cell. */
```

Comment:
Can you please explain a bit more in this comment why we need to handle the timeout case specially? Don't we already have code that removes expired intro circuits from the circuitmap etc.? Why do we need more handling code for this?

Reply:
Added in fixup commit: ce79052eab

Reply:
LGTM!